docs/conf.py

```diff
@@ -238,7 +241,91 @@
                       'issue '),
             'fileref': ('https://github.com/NVIDIA/DALI/tree/' + (git_sha if git_sha != u'0000000' else "master") + '/%s', ''),}
 
+
+from typing import (
+    Any, Callable, Dict, Iterator, List, Optional, Sequence, Set, Tuple, Type, TypeVar, Union
+)
+from typing import get_type_hints
+
+
+_dali_enums = ["DALIDataType", "DALIIterpType", "DALIImageType", "PipelineAPIType"]
+
+class EnumDocumenter(ClassDocumenter):
+    # Register as .. autoenum::
+    objtype = 'enum'
+    # Produce .. py:class:: fields in the RST doc
+    directivetype = 'class'
+    # priority = 10
+
+    def __init__(self, *args: Any) -> None:
+        super().__init__(*args)
+
+    @classmethod
+    def can_document_member(cls, member: Any, membername: str, isattr: bool, parent: Any
+                            ) -> bool:
+        """Verify that we handle only the registered DALI enums. Pybind doesn't subclass Enum class,
+        so we need an explicit list.
+        """
+        return membername in _dali_enums and isinstance(parent, ClassDocumenter)
+
+    def filter_members(self, members, want_all):
+        """After self.get_object_members() obtained all members, this function filters only
+        the ones we're interested in.
+        We can do the sorting here based on the values, and pass through in self.sort_members()
+        """
+        filtered = super().filter_members(members, want_all)
+
+        # sort by the actual value of enum - this is a tuple of (name, value, boolean)
+        def get_member_value(member_desc):
+            _, member_value, _ = member_desc
+            if isinstance(member_value, Enum):
+                return member_value.value
+            else:
+                return int(member_value)
+        filtered.sort(key = get_member_value)
+
+        return filtered
+
+    def sort_members(self, documenters, order):
+        """Ignore the order. Here we have access only to documenters that carry the name
+        and not the object. We need to sort based on the enum values and we du it in
+        self.filter_members()
+        """
+        return documenters
+
+class EnumAttributeDocumenter(AttributeDocumenter):
+    # Give us higher priority over Sphinx native AttributeDocumenter
+    priority = 12
```

Comment:
Why not 14?

Reply:
It can be anything bigger than 10 (AttributeDocumenter priority) or 11 (more specialized like InstanceAttributeDocumenter, though I'm not sure if we need to be higher than them).
https://github.com/sphinx-doc/sphinx/blob/7db3633778a24c4fc90c8a0ceb2318d6df7cf4e3/sphinx/ext/autodoc/__init__.py#L2027

So I've picked 12. I can make it 42 if you want.

Reply:
12 sound good in that case. Just please add more comment how it was picked and how to adjust if anything changes in Sphinx.